### ISSUE TITLE
Devfile registry operator pre-release to OperatorHub

### ISSUE THREAD

**Issue by michael-valdron** [devfile/api#1211]
## Which area/kind this issue is related to?

/area registry
/area releng

## Issue Description

In #410, we plan to make our first release of the registry operator available on [OperatorHub](https://operatorhub.io) for consumers. To ensure early adopters are ready to ship when this release becomes available, it would be necessary to also provide pre-releases.

Resolve devfile/api#1106 to ensure no problems with publishing.

## Acceptance Criteria

- [x] Required documentation for publication https://k8s-operatorhub.github.io/community-operators/packaging-operator/#provide-information-about-your-operator
    - [x] What the managed application is about and where to find more information
    - [x] The features of your Operator and how to use it
    - [x] Any manual steps required to fulfill pre-requisites for running/installing your Operator
- [x] Cut pre-release number for registry operator
- [x] Ensure bundle files are up to date and placed under the following directory structure within a folk of [k8s-operatorhub/community-operators](https://github.com/k8s-operatorhub/community-operators): https://k8s-operatorhub.github.io/community-operators/packaging-operator/#create-a-release
- [x] Create a PR of these change to the folk within the [k8s-operatorhub/community-operators](https://github.com/k8s-operatorhub/community-operators) repository and merge once approved

**Comment by michael-valdron** [comment]
> Cut pre-release number for registry operator

I think we should start with `0.1.0`, increment on minor until our first release `1.0.0`.

**Comment by michael-valdron** [comment]
> Add minimal documentation for publication to OperatorHub

Documentation acceptance criteria has been included.

**Comment by michael-valdron** [comment]
Begun updating manifests and set version fields to `0.1.0`.

**Comment by michael-valdron** [comment]
Waiting for devfile/registry-operator#45 to merge: https://github.com/devfile/api/issues/1106#issuecomment-1699848699

**Comment by michael-valdron** [comment]
Rebased prep changes with recent changes from https://github.com/devfile/registry-operator/pull/45 & https://github.com/devfile/registry-operator/pull/46

**Comment by michael-valdron** [comment]
Ran into a problem within the registry operator, it seems that `kustomize` has an ongoing bug which prevents us from using it to build off custom resources which `kind` ends with `List`: https://github.com/kubernetes-sigs/kustomize/issues/5042

For this issue, I cannot generate the examples of `ClusterDevfileRegistriesList` and `DevfileRegistriesList` under the CSV, I can continue without these and open a new issue for this separately to fix once https://github.com/kubernetes-sigs/kustomize/issues/5042 is resolved.

Will try manually entering the examples of `ClusterDevfileRegistriesList` and `DevfileRegistriesList` at this time.

**Comment by michael-valdron** [comment]
It seems that `kustomize` has an ongoing bug which prevents us from using it to build off custom resources which `kind` ends with `List`: https://github.com/kubernetes-sigs/kustomize/issues/5042

For this issue, I cannot generate the examples of `ClusterDevfileRegistriesList` and `DevfileRegistriesList` under the CSV, I can continue without these and open a new issue for this separately to fix once https://github.com/kubernetes-sigs/kustomize/issues/5042 is resolved.

Will try manually entering the examples of `ClusterDevfileRegistriesList` and `DevfileRegistriesList` at this time.

**Comment by michael-valdron** [comment]
Part 1 of this issue is ready for review: https://github.com/devfile/registry-operator/pull/47

**Comment by michael-valdron** [comment]
Part 2: Cutting the new release `v0.1.0` in the registry operator [repository](https://github.com/devfile/registry-operator)
Part 3: Begin the process of publication to OperatorHub by opening a PR to [k8s-operatorhub/community-operators](https://github.com/k8s-operatorhub/community-operators) with the version bundle

**Comment by michael-valdron** [comment]
New release `v0.1.0` has been cut.

> Part 3: Begin the process of publication to OperatorHub by opening a PR to [k8s-operatorhub/community-operators](https://github.com/k8s-operatorhub/community-operators) with the version bundle

This part is being blocked by #1251 due to OLM validation passing being required for publication to OperatorHub.

**Comment by michael-valdron** [comment]
Contributing doc for OperatorHub: https://github.com/operator-framework/community-operators/blob/master/docs/contributing-via-pr.md

**Comment by rm3l** [comment]
Candidate for demo at the end of Sprint 2247

**Comment by michael-valdron** [comment]
https://github.com/devfile/registry-operator/pull/49#issuecomment-1724086442

**Comment by michael-valdron** [comment]
Keeping status blocked to include #1028 under new release cut.

**Comment by michael-valdron** [comment]
New release [v0.1.1](https://github.com/devfile/registry-operator/releases/tag/v0.1.1) has been cut on GitHub.

> Part 3: Begin the process of publication to OperatorHub by opening a PR to k8s-operatorhub/community-operators with the version bundle

Starting part 3 to publish to OperatorHub.

**Comment by michael-valdron** [comment]
Opened a PR with current version's bundle: 
https://github.com/k8s-operatorhub/community-operators/pull/3287


Experiencing an failing check which seems to be flagging a missing field which is not missing from the CSV file: https://github.com/k8s-operatorhub/community-operators/pull/3287#discussion_r1336452067

**Comment by michael-valdron** [comment]
> Opened a PR with current version's bundle: [k8s-operatorhub/community-operators#3287](https://github.com/k8s-operatorhub/community-operators/pull/3287)
> 
> Experiencing an failing check which seems to be flagging a missing field which is not missing from the CSV file: [k8s-operatorhub/community-operators#3287 (comment)](https://github.com/k8s-operatorhub/community-operators/pull/3287#discussion_r1336452067)

Opened Q&A discussion about this problem: https://github.com/k8s-operatorhub/community-operators/discussions/3294

**Comment by michael-valdron** [comment]
> > Opened a PR with current version's bundle: [k8s-operatorhub/community-operators#3287](https://github.com/k8s-operatorhub/community-operators/pull/3287)
> > Experiencing an failing check which seems to be flagging a missing field which is not missing from the CSV file: [k8s-operatorhub/community-operators#3287 (comment)](https://github.com/k8s-operatorhub/community-operators/pull/3287#discussion_r1336452067)
> 
> Opened Q&A discussion about this problem: [k8s-operatorhub/community-operators#3294](https://github.com/k8s-operatorhub/community-operators/discussions/3294)

**Update**: I received feedback on this problem and found that `categories` was misspelled under the CSV :facepalm:. I have corrected this in https://github.com/devfile/registry-operator/pull/52 and will cut a new patch version for the OperatorHub publication.

**Comment by michael-valdron** [comment]
There is a missing place to update the operator version: https://github.com/k8s-operatorhub/community-operators/pull/3287#discussion_r1349245313

Investigating this on the operator side.

**Comment by michael-valdron** [comment]
> There is a missing place to update the operator version: [k8s-operatorhub/community-operators#3287 (comment)](https://github.com/k8s-operatorhub/community-operators/pull/3287#discussion_r1349245313)
> 
> Investigating this on the operator side.

Opened PR for new patch changes: https://github.com/devfile/registry-operator/pull/54

These changes will fully pass kiwi testing: https://github.com/k8s-operatorhub/community-operators/actions/runs/6487040924/job/17616540337

**Comment by michael-valdron** [comment]
PR to publish to OperatorHub.io is now ready for review: https://github.com/k8s-operatorhub/community-operators/pull/3287

**Comment by michael-valdron** [comment]
Publication PR https://github.com/k8s-operatorhub/community-operators/pull/3287 has now been merged and the registry operator is now published to OperatorHub.io: https://operatorhub.io/operator/registry-operator

Closing this issue.
